components/forms/CreateRoleForm.tsx

```diff
@@ -1,45 +1,64 @@
-import { Form, InputNumber, Input, Button } from 'antd';
+import { Form, InputNumber, Input, Modal } from 'antd';
 import rolesApi from '../../api/rolesApi';
-import { RolePostData } from '../../types/role';
+import { CompanyListData } from '../../types/company';
+import { RoleData } from '../../types/role';
+import { Nullable } from '../../types/utils';
+import { DEFAULT_NUM_MONTHS_BEFORE_ROLE_START } from '../../utils/constants';
+import CompanyOption from '../CompanyOption';
 import RoleTypesSelect from './RoleTypesSelect';
 
-function CreateRoleForm() {
+type Props = {
+  company: Nullable<CompanyListData>;
+
+  isOpen: boolean;
+  closeForm: () => void;
+  onCreate: (role: RoleData) => void;
+};
+
+function getDefaultRoleYear(): number {
+  const date = new Date();
+  date.setMonth(date.getMonth() + DEFAULT_NUM_MONTHS_BEFORE_ROLE_START);
+  return date.getFullYear();
+}
+
+function CreateRoleForm({ company, isOpen, closeForm, onCreate }: Props) {
   const [form] = Form.useForm();
 
-  const onSubmit = (values: RolePostData) => {
-    rolesApi
-      .createRole(values)
-      .then(() => {
+  const onSubmit = (companyId: number) => {
+    form.validateFields().then(({ title, type, year }) => {
+      rolesApi.createRole({ companyId, title, type, year }).then((resp) => {
+        if (resp.payload.id === undefined) {
+          return;
+        }
+
         form.resetFields();
-        alert('Role created successfully!');
-      })
-      .catch((error) => alert(error)); // temporarily while we set up error displays
+        closeForm();
+        onCreate(resp.payload as RoleData);
+      });
+    });
   };
 
+  if (company === null) {
```

Comment:
The problem that is causing the bug above. Might want to add a notification.

Reply:
Not really a bug and this is intended. How would the notification work? I think we don't want the role modal to be opened if company is not filled.

Reply:
Yes. I think disabling the role field when the company is not filled is a good idea. However, we need to let the user know why it is disabled somehow.